src/utils/placeholder.ts

```diff
@@ -0,0 +1,28 @@
+import { Domain, DomainPriority } from '../external';
+import { newDomain } from './domains';
+
+/**
+ * Provide a placeholder domain so that we can test for validity later, but
+ * it can be safely operated on like a real domain.
+ */
+export const placeholder = (min: number, max: number): Domain =>
+  newDomain(min, max, DomainPriority.PLACEHOLDER);
+
+export const withoutPlaceholder = (
```

Comment:
It seems everywhere `withoutPlaceholder` is used we return placeholder as a default value (i.e. `withoutPlaceholder(c.xDomain) || PLACEHOLDER_DOMAIN`), shouldn't it be renamed to something like `withPlaceholder` and take in placeholder as an argument?